plugin: Fix how we send the shared_secret to the plugin in htlc_accepted.

cdecker opened the pull request:
~Found this while trying to do some magic on top of `shared_secret`, turns out we were not keeping the secret around long enough, so now we just stash it in the `htlc_accepted_payload`.~

I had confused pointer and pointee :frowning_face:

rustyrussell commented:
Why did valgrind not catch this BTW?

cdecker commented:
Sorry, my bad, the usual confusion with pointer and pointee :frowning_face:

rustyrussell commented on lightningd/peer_htlcs.c:
```diff
@@ -721,7 +721,7 @@ static void htlc_accepted_hook_serialize(struct htlc_accepted_hook_payload *p,
 
 	json_add_hex_talarr(s, "next_onion", p->next_onion);
-	json_add_hex(s, "shared_secret", &hin->shared_secret, sizeof(hin->shared_secret));
+	json_add_hex(s, "shared_secret", hin->shared_secret, sizeof(struct secret));
```
Gah, this is why writing helpers is important!

But yes, as @ZmnSCPxj points out, the commit msg is now nonsensical.

ZmnSCPxj replied:
So new `json_add_secret`?

cdecker commented:
Right, I shouldn't rush these things, my bad.
 - Added a wrapper to add `struct secret` to a JSON stream
 - Using that wrapper now
 - Fixed commit messages.

Sorry for the confusion :slightly_frowning_face:
